documentation update (#170)
ltelab/disdrodb: CONTRIBUTING.rst

to identify simple code quality issues. When an issue is identified
(the pre-commit script exits with non-zero status), the hook aborts the
commit and prints the error. Currently, DISDRODB only tests that the
code to be committed complies with black's format style and the ruff linter.

In case that the commit is aborted, you only need to run black agains you code.
This can be done by running   ``black .``  or   ``ruff check .``

.. note::

> 	To maintain consitency, please use version and configuration defined into `.pre-commit-config.yaml`).

Reviewer: ```suggestion
	To maintain consistency, please use version and configuration defined into `.pre-commit-config.yaml`.
```

The can also be done with  ``pre-commit run --all-files``. This is recommended since it
indicates if the commit contained any formatting errors (that are automatically corrected).

> More info on pre-commit and CI tools in the Code quality ans testing section  `Code quality and testing section <#code-quality-control-and-testing>`__

Reviewer: ```suggestion
More info on pre-commit and CI tools are provided in the Code quality and testing section  `Code quality and testing section <#code-quality-control-and-testing>`__
```

Code quality control and testing
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Every code change must be tested !

To maintain a high code quality, we use `pre-commit <https://pre-commit.com/>`__ to run a series of checks on the code before each commit. The checks are defined in the .pre-commit-config.yaml file. The checks include:

> 

Reviewer: precommit does not include testing and coverage stuff !
Split precommit stuffs from `"testing and code coverage tool"

+-----------------------------------------------------------------------------------------------+------------------------------------------------------------------+------------+-------+-------------------------------------------+
| Coverage                                                                                      | Measure the code coverage of the project's unit tests            | -          | 👍    | all versions according to GitHub workflow |
+-----------------------------------------------------------------------------------------------+------------------------------------------------------------------+------------+-------+-------------------------------------------+
| `CodeCov    <https://about.codecov.io/>`__                                                    | Uses the "coverage" package to generate a code coverage report.  | -          | 👍    | all versions according to GitHub workflow |
+-----------------------------------------------------------------------------------------------+------------------------------------------------------------------+------------+-------+-------------------------------------------+

> | `Coveralls    <https://coveralls.io/>`__                                                      | Uses the "coverage" to track the quality of your code over time. | -          | 👍    | all versions according to GitHub workflow |

Reviewer: @regislon you forgot to copy CodeCov and Coveralls in the table below ;)

| `CodeFactor <https://www.codefactor.io/>`__                                                   | Automated code review and analysis tools                         | -          | 👍    | ?                                         |
+-----------------------------------------------------------------------------------------------+------------------------------------------------------------------+------------+-------+-------------------------------------------+

Code testing
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Every code change must be tested !

**Pytest**

DISDRODB tests are written using the third-party `pytest <https://docs.pytest.org>`_ package.

The tests located in the `disdrodb/tests` folder are used to test various functions of the code and are automatically run when changes are pushed to the main repository through a GitHub Pull Request.

.. code-block:: bash

	pytest disdrodb/tests

To create a new reader test, simply add a small, single-station dataset and the associated files (issue, metadata), and expected data, in the following manner:

| 📁 disdrodb
| ├── 📁 tests
|     	├── 📁 pytest_files

> |           ├── 📁 check_readers

Reviewer: I wonder if we don't want to shortcut that to `data/readers/` or create disdrodb/test_data directory to avoid too long fpaths

Reviewer: `disdrodb/test/pytest_files/check_readers` vs `disdrodb/test_data/readers`

|     	      ├── 📁 DISDRODB
|     		      ├── 📁 Raw
|     			      ├── 📁 `<data_source>` : e.g. GPM, ARM, EPFL, ...
|     				      ├── 📁 `<campaign_name>` : e.g. PARSIVEL_2007
|     				         ├── 📁 data
|     				            ├── 📁 `<station_name>`.\*
|     				         ├── 📁 issue
|     				            ├── 📁 `<station_name>`.yml
|     				         ├── 📁 metadata
|     				            ├── 📁 `<station_name>`.yml
|     				         ├── 📁 ground_truth
|     				            ├── 📁 `<station_name>`.\*

> A single test will run all readers with data that has been placed in the above-mentioned structure. The raw data will be processed, and the resulting Apache Parquet files (l0a) will be compared to the ground truth.

Reviewer: ```suggestion
A single test will run all readers with data that has been placed in the above-mentioned structure. The raw data will be processed, and the resulting Apache Parquet files (L0A) will be compared to the ground truth.
```

	pytest disdrodb

> The tests located in the `disdrob/tests` folder are used to test various functions of the code and are automatically run when changes are pushed to the main repository through a GitHub Pull Request.

Reviewer: ```suggestion
The tests located in the `disdrodb/tests` folder are used to test various functions of the code and are automatically run when changes are pushed to the main repository through a GitHub Pull Request.
```

**Black**

Black should be used that way :

.. code-block:: bash

	black .

.. note::

> 	To maintain consitency, make sure to stick to version defined in the `.pre-commit-config.yaml` file. This version will be used in the CI.

Reviewer: ```suggestion
	To maintain consistency, make sure to stick to the version defined in the `.pre-commit-config.yaml` file. This version will be used in the CI.
```

During this process, pre-commit hooks will be run. Your commit will be
allowed only if quality requirements are fulfilled.

> If you encounter errors, Black ad Ruff can be run using the following command :

Reviewer 2: ```suggestion
If you encounter errors, Black and Ruff can be run using the following command :
```
